packages/strapi-design-system/src/v2/SimpleMenu/SimpleMenu.tsx

```diff
@@ -0,0 +1,84 @@
+import { useRef, useState } from 'react';
+
+import * as Menu from './Menu';
+import { useId } from '../../hooks/useId';
+import { useIntersection } from '../../hooks/useIntersection';
+
+/* -------------------------------------------------------------------------------------------------
+ * SimpleMenu
+ * -----------------------------------------------------------------------------------------------*/
+
+interface SimpleMenuProps
+  extends Omit<Menu.TriggerProps, 'children'>,
+    Pick<Menu.ContentProps, 'popoverPlacement' | 'intersectionId'> {
+  children?: React.ReactNode;
+  id?: string;
+  label?: React.ReactNode;
+  onOpen?: () => void;
+  onClose?: () => void;
+  /**
+   * Callback function to be called when the popover reaches the end of the scrollable content
+   */
+  onReachEnd?: (entry: IntersectionObserverEntry) => void;
+}
+
+const SimpleMenu = ({
+  children,
+  id,
+  onOpen = () => {},
+  onClose = () => {},
+  popoverPlacement,
+  onReachEnd,
+  ...props
+}: SimpleMenuProps) => {
+  /**
+   * Used for the intersection observer
+   */
+  const contentRef = useRef<HTMLDivElement>(null);
+
+  const [internalIsOpen, setInternalIsOpen] = useState(false);
+
+  const handleReachEnd = (entry: IntersectionObserverEntry) => {
+    if (onReachEnd) {
+      onReachEnd(entry);
+    }
+  };
+
+  const handleOpenChange = (isOpen: boolean) => {
+    if (isOpen) {
+      onOpen();
+    } else {
+      onClose();
+    }
```

Comment:
Should we do a typecheck here because both props are optional?

Reply:
no because there are `noop` defaults declared.

Reply:
Kind of: you recently commented on one of my PRs that onClose could also be something different, e.g. `null` and in this case it would throw. I think you had a good point there it is a good idea to wrap functions always in a type-check.

Reply:
aha great point, to be honest i prefer stricter checking than defaults anyway 😛